cve_bin_tool/util.py

```diff
@@ -280,24 +280,21 @@ def __missing__(self, key: str) -> list[CVE] | set[str]:
 
 def regex_find(
     lines: str, version_patterns: list[Pattern[str]], ignore: list[Pattern[str]]
-) -> list[str]:
+) -> set[str]:
     """Search a set of lines to find all matches for the given regex"""
-    versions = list()
+    versions = set()
 
     for pattern in version_patterns:
         version_matches = pattern.finditer(lines)
         for match in version_matches:
-            # before collecting a potential version number, ensure the version string isn't on the ignore list
-            if not check_ignored(match.string, ignore):
+            matched_text = match.group(0)
+            if not check_ignored(matched_text, ignore):
                 version = match.group(1).strip()
                 version = version.replace("_", ".").replace("-", ".")
-                versions.append(version)
+                versions.add(version)
 
-    # if we searched and found no matches, just return a one-element list containing "UNKNOWN"
-    if not versions:
-        versions.append("UNKNOWN")
-
-    return versions
+    # return unknown if none found
```

Comment:
Same here; I think it's worth being clear that we're still returning a list/set, but it has only one element and the element is "UNKNOWN".